### PR TITLE
Make promptYesOrNo take defaultInput rather than defaultResult

### PR DESCRIPTION
**Pull request by idrysdale**
* Print the options depending on defaultValue passed to `promptYesOrNo`

### CODE REVIEW

File: fluidkeys/main.go [hunk 1]
```diff
@@ -378,12 +378,19 @@ func promptForKeyToImportFromGpg(secretKeyListings []gpgwrapper.SecretKeyListing
 	}
 }
 
-func promptYesOrNo(message string, defaultResult bool) bool {
+func promptYesOrNo(message string, defaultInput string) bool {
 	for {
-		input := promptForInput(message)
+		var options string
+		switch strings.ToLower(defaultInput) {
+		case "y":
```
Comment (paulfurley) [inline fluidkeys/main.go]: clever!!

File: fluidkeys/main.go [hunk 2]
```diff
@@ -394,23 +401,28 @@ func promptYesOrNo(message string, defaultResult bool) bool {
...
+	if input != "" {
+		return input
+	} else {
+		return defaultInput
+	}
 }
 
-func promptForInput(prompt string) string {
-	return promptForInputWithPipes(prompt, bufio.NewReader(os.Stdin))
+func promptForInput(prompt string, defaultInput string) string {
```
Comment (paulfurley) [inline fluidkeys/main.go]: I'm not sure the value of adding a `defaultInput` to this generic `promptForInput`function? 

AFAICT there are no callers `promptForInput` that want to specify a default input, so they all end up having to add `""` to their argument list, which is weird

Comment (idrysdale) [reply fluidkeys/main.go]: Agreed, I'll move it down a level into `promptForInputWithPipes` and call that directly from `promptYesOrNo`

Comment (paulfurley) [reply fluidkeys/main.go]: I think that' still overcomplicating things. Why not do this logic in `promptYesOrNo` ? that's the only place that the default makes sense

**Pull request merged**